bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/builders/AutomationChannelBuilder.java

```diff
@@ -0,0 +1,244 @@
+import org.openhab.binding.hdpowerview.internal.HDPowerViewTranslationProvider;
+import org.openhab.binding.hdpowerview.internal.api.responses.SceneCollections.SceneCollection;
+import org.openhab.binding.hdpowerview.internal.api.responses.Scenes.Scene;
+import org.openhab.binding.hdpowerview.internal.api.responses.ScheduledEvents;
+import org.openhab.binding.hdpowerview.internal.api.responses.ScheduledEvents.ScheduledEvent;
+import org.openhab.core.library.CoreItemFactory;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelGroupUID;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AutomationChannelBuilder} class creates automation channels
+ * from structured scheduled event data.
+ *
+ * @author Jacob Laursen - Initial contribution
+ */
+@NonNullByDefault
+public class AutomationChannelBuilder {
+
+    private final Logger logger = LoggerFactory.getLogger(AutomationChannelBuilder.class);
+    private final HDPowerViewTranslationProvider translationProvider;
+    private final ChannelGroupUID channelGroupUid;
+    private final ChannelTypeUID channelTypeUid = new ChannelTypeUID(HDPowerViewBindingConstants.BINDING_ID,
+            HDPowerViewBindingConstants.CHANNELTYPE_AUTOMATION_ENABLED);
+
+    private List<Channel> channels;
+    private Map<Integer, Scene> scenes;
+    private Map<Integer, SceneCollection> sceneCollections;
+    private List<ScheduledEvent> scheduledEvents;
+
+    public AutomationChannelBuilder(HDPowerViewTranslationProvider translationProvider,
+            ChannelGroupUID channelGroupUid) {
+        this.translationProvider = translationProvider;
+        this.channelGroupUid = channelGroupUid;
+        this.channels = new ArrayList<>(0);
+        this.scheduledEvents = new ArrayList<>(0);
+        this.scenes = new HashMap<>(0);
+        this.sceneCollections = new HashMap<>(0);
+    }
+
+    /**
+     * Creates an {@link AutomationChannelBuilder} for the given {@link HDPowerViewTranslationProvider} and
+     * {@link ChannelGroupUID}.
+     * 
+     * @param translationProvider the {@link HDPowerViewTranslationProvider}
+     * @param channelGroupUid parent {@link ChannelGroupUID} for created channels
+     * @return channel builder
+     */
+    public static AutomationChannelBuilder create(HDPowerViewTranslationProvider translationProvider,
+            ChannelGroupUID channelGroupUid) {
+        return new AutomationChannelBuilder(translationProvider, channelGroupUid);
+    }
+
+    /**
+     * Adds created channels to existing list.
+     * 
+     * @param channels list that channels will be added to
+     * @return channel builder
+     */
+    public AutomationChannelBuilder withChannels(List<Channel> channels) {
+        this.channels = channels;
```

Comment:
And then use ..

```
this.channels.clear();
this.channels.addAll(channels);
```

Reply:
I want to be able to use existing list without copying all contents.

Reply:
And then the .withXX() method could assign the new list...